libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/AppConfigService.java

```diff
@@ -0,0 +1,122 @@
+import software.amazon.awssdk.services.secretsmanager.model.ResourceNotFoundException;
+import software.amazon.lambda.powertools.parameters.ParamManager;
+import software.amazon.lambda.powertools.parameters.SecretsProvider;
+import uk.gov.di.ipv.core.library.annotations.ExcludeFromGeneratedCoverageReport;
+import uk.gov.di.ipv.core.library.config.EnvironmentVariable;
+import uk.gov.di.ipv.core.library.helpers.LogHelper;
+
+import java.io.IOException;
+import java.util.List;
+
+import static java.time.temporal.ChronoUnit.MINUTES;
+import static uk.gov.di.ipv.core.library.config.EnvironmentVariable.CONFIG_SERVICE_CACHE_DURATION_MINUTES;
+import static uk.gov.di.ipv.core.library.config.EnvironmentVariable.ENVIRONMENT;
+import static uk.gov.di.ipv.core.library.helpers.LogHelper.LogField.LOG_SECRET_ID;
+
+public class AppConfigService extends YamlParametersConfigService {
+
+    private static final Logger LOGGER = LogManager.getLogger(AppConfigService.class);
+    private static final int DEFAULT_CACHE_DURATION_MINUTES = 3;
+    private static final String CORE_BASE_PATH = "/%s/core/";
+    private static final String CORE = "core";
+
+    @Getter @Setter private List<String> featureSet;
+    private final SecretsProvider secretsProvider;
+
+    @ExcludeFromGeneratedCoverageReport
+    public AppConfigService() {
+        var cacheDuration =
+                getEnvironmentVariable(
+                        CONFIG_SERVICE_CACHE_DURATION_MINUTES, DEFAULT_CACHE_DURATION_MINUTES);
+        var paramsRaw = getRawParams(cacheDuration);
+
+        initializeConfig(paramsRaw);
+        this.secretsProvider =
+                ParamManager.getSecretsProvider(
+                                SecretsManagerClient.builder()
+                                        .httpClient(UrlConnectionHttpClient.create())
+                                        .build())
+                        .defaultMaxAge(cacheDuration, MINUTES);
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    public AppConfigService(String paramsRaw, SecretsProvider secretsProvider) {
+        initializeConfig(paramsRaw);
+        this.secretsProvider = secretsProvider;
+    }
+
+    private void initializeConfig(String paramsRaw) {
+        try {
+            var paramsYaml = YAML_OBJECT_MAPPER.readTree(paramsRaw).get(CORE);
+            addJsonConfig(parameters, paramsYaml);
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Could not load parameter yaml", e);
+        }
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    private String getRawParams(Integer cacheDuration) {
+        var applicationId = getEnvironmentVariable(EnvironmentVariable.APP_CONFIG_ID);
+        var environmentId = getEnvironmentVariable(EnvironmentVariable.APP_CONFIG_ENVIRONMENT_ID);
+        var profileId = getEnvironmentVariable(EnvironmentVariable.APP_CONFIG_PROFILE_ID);
+
+        return ParamManager.getAppConfigProvider(
```

Comment:
Not sure if you're aware of [this bug](https://github.com/aws-powertools/powertools-lambda-java/issues/1672) in the default `AppConfigProvider` ... they accepted my PR for the fix, and have merged it, but it has yet to make it into a release 🙈 

Not sure if you doing anything clever elsewhere (maybe in your custom client) to work around the problem?

Reply:
Once per execution or once per lambda lifetime? If the former, then you may still hit the bug ... I think.